### DISCUSSION TITLE
Add a chunk in the NCReader

### DISCUSSION THREAD

**Discussion by kongdd** [Deltares/Wflow.jl#441] [Feature request]
Suggest adding a chunk in the struct of NCReader.
In my running case (10 years hourly runoff simulation), 90% time consumed at loading forcing data.

Adding a chunk (with the dimension of `nlon*nlat*chunksize`, chunksize=1000) to store forcing data should reduce the nc reading time significantly.

If the running date is still within the current chunk, read data from the chunk; otherwise, update the chunk and read again.

https://github.com/Deltares/Wflow.jl/blob/6a28144298831bb2d038d04e67e4519c5f31e22e/src/io.jl#L623-L630

**Comment by verseve** [comment]
Thanks for the suggestion. We generally run with forcing data that has a `chunksize` of 1 (`time` dimension), then ~2% of total simulation time is used for loading forcing (and cyclic) data (daily time step). 

Do you see a reduction in time consumption when a `chunksize` of 1 is used in the forcing data? 

If not, it would be good to know the Wflow.jl version (and preferably the NCDatasets version) used, Operating System and ideally a test case (part of your running case).

**Reply by kongdd** [reply]
Thanks for your solution. The running times is reduced about 10 times.

`setting chunksize to 1 (time dimension)` might need to be documented, which will benefit other users.
